pandas/io/html.py

```diff
@@ -952,7 +1005,19 @@ def _parse(flavor, io, match, attrs, encoding, displayed_only, **kwargs):
     ret = []
     for table in tables:
         try:
-            ret.append(_data_to_frame(data=table, **kwargs))
+            df = _data_to_frame(data=table, **kwargs)
+            # Cast MultiIndex header to an Index of tuples when extracting header
+            # links and replace np.nan with None.
+            # This maintains consistency of selection (e.g. df.columns.str[1])
+            if extract_links in ("all", "header"):
+                idx = df.columns.values
```

Comment:
Hmm could you do something like

```
new_idx = Index([tuple(None if np.isnan(col) else col for col in df.columns], dtype=object, tupleize_cols=False)
df.columns = new_idx
```

instead.

Reply:
Thanks for the suggestion! I tried this, but it did not work. `np.isnan` has to be run vectorized, and even when I switched to pd.isna the nans stayed as they were.

Reply:
Actually, this set me on the right track for a working fix, so thank you again!